bot/exts/moderation/infraction/infractions.py

```diff
@@ -509,6 +503,34 @@ async def apply_ban(
         async def action() -> None:
             await ctx.guild.ban(user, reason=reason, delete_message_days=purge_days)
 
+        # If user has an elevated role (staff, partner, or community), require
+        # confirmation before banning.
+        if isinstance(user, Member) and any(role.id in constants.STAFF_PARTNERS_COMMUNITY_ROLES for role in user.roles):
+            confirmation_view = _utils.StaffBanConfirmationView(
+                allowed_users=(ctx.author.id,),
+                allowed_roles=constants.MODERATION_ROLES,
+                timeout=10,
+            )
+            confirmation_view.message = await ctx.send(
+                f"{user} has an elevated role. Are you sure you want to ban them?",
+                view=confirmation_view
+            )
+
+            timed_out = await confirmation_view.wait()
+            if timed_out:
+                log.trace(f"Attempted ban of user {user} by moderator {ctx.author} cancelled due to timeout.")
+                return None
+
+            if confirmation_view.confirmed is False:
+                log.trace(f"Attempted ban of user {user} by moderator {ctx.author} cancelled due to manual cancel.")
+                return None
+
+        infraction = await _utils.post_infraction(ctx, user, "ban", reason, active=True, **kwargs)
```

Comment:
Moving this below
```
        if reason:
            reason = textwrap.shorten(reason, width=512, placeholder="...")
```
causes a change in behaviour.

I assume the behaviour we want is for the full message to be sent to the database, but the reason to be trimmed in the audit log (I assume because there's a limit there), but now it's also trimmed in the database.

Reply:
We should modify the tests to catch this too https://github.com/python-discord/bot/blob/36a9ac91f51a3507d1609c899ccf54bb4c0f3dc6/tests/bot/exts/moderation/infraction/test_infractions.py#L16

You can do that here if you want otherwise I can open an issue for it.

Reply:
See #2955.